WordPress/src/main/java/org/wordpress/android/viewmodel/giphy/provider/TenorProvider.kt

```diff
@@ -23,7 +29,8 @@ import org.wordpress.android.viewmodel.giphy.provider.GifProvider.GifRequestFail
 
 internal class TenorProvider @JvmOverloads constructor(
     val context: Context,
-    private val tenorClient: IApiClient
+    private val tenorClient: IApiClient,
+    private val scope: CoroutineScope = CoroutineScope(Dispatchers.Default)
```

Comment:
Do you think it makes sense to make an actual use of the scope and make the whole provider aware of the lifecycle? Maybe with a `cancel` method? This is not necessary, I'm just wondering about what you think about it 👍

Reply:
After your review, I've decided to completely remove the coroutine usage since the manual timeout handling wasn't necessary anymore and given the fact we should let the thread handling be done internally. So right now the TenorProvider no longer receives a `CoroutineScope`. I've committed this in 9bc5cbb4d3b13d7f01e4b317f69fcfb09a1cb8fa.

What do you think? Should we go for the lifecycle awareness even so?